### ISSUE TITLE
Improvement in error message when a property value is assigned the wrong type

### ISSUE THREAD

**Issue by Dr-Irv** [microsoft/pyright#3007]
**Is your feature request related to a problem? Please describe.**
If you have a property in a class, and you do an assignment of the wrong type, the error message from pyright doesn't help you figure out that was the problem.

**Describe the solution you'd like**
A better error message.

**Additional context**
Consider the following simple example:
```python
class PTest:
    @property
    def x(self) -> int:
        return self._x

    @x.setter
    def x(self, xval: int):
        self._x = xval


p = PTest()
p.x = "hey"

y: int = 10
y = "abc"
```

For the line assigning `p.x`, the message is:
```
Cannot assign member "x" for type "PTest"
```
For the line assigning `y`, the message is:
```
Expression of type "Literal['abc']" cannot be assigned to declared type "int"
  "Literal['abc']" is incompatible with "int"
```

I spent an hour with a colleague figuring out that the type we assigned to a property was wrong because the error message gave us no hint.

For the case of `p.x`, can we get a message that looks like the message for assigning `y` where it points out the type mismatch?

**Comment by erictraut** [comment]
Thanks for the suggestion. I agree that this error message is not as descriptive as it could be. This affects not only properties but descriptor objects. I've improved the error message for the `__set__` method in general to include additional information. In the case where the `__set__` method is not overloaded (which is typical), the message now looks like this:

```
Cannot assign member "x" for type "PTest"
  "Literal['hey']" is incompatible with "int"
```

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.220, which I just published. It will also be included in the next release of pylance.
